server/controllers/profile.js

```diff
@@ -0,0 +1,102 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/create
+// @desc create a profile
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const {
+        userId,
+        firstName,
+        lastName,
+        description,
+        gender,
+        address,
+        birthDate,
+        email,
+        phoneNumber,
+        isAvailable,
+        availability
+    } = req.body
+    
+    const profile = await Profile.create({
+        userId,
+        firstName,
+        lastName,
+        description,
+        gender,
+        birthDate,
+        email,
+        phoneNumber,
+        address,
+        isAvailable,
+        availability,
+    })
+
+    if(!profile) {
+        res.status(400);
+        throw new Error("Invalid request data");
+    }
+    
+     res.status(201).json({
+        success: true,
+        data: profile
```

Comment:
Good job specifying the creation status!

Reply:
Good on using the right status, I don't see the need to pass in a success boolean and a data payload property, I would just return the profile data `res.status(201).json({ profile })` 

It's simpler, we can validate on the front-end whether a request was successful indicated by the 2xx HTTP status codes.